lightning_server.cc

```diff
@@ -33,32 +31,16 @@ void LightningServer::start() {
   acceptor_.bind(endpoint);
   acceptor_.listen();
 
+  // Lightning listening loop
   for (;;) {
     // Accept connection request
-    boost::asio::ip::tcp::socket socket(io_service_);
-    acceptor_.accept(socket);
-
-    // Read in request
-    char request_buffer[MAX_REQ_SIZE];
-    boost::system::error_code ec;
-    std::size_t bytes_read = socket.read_some(boost::asio::buffer(request_buffer), ec);
-    std::cout << "~~~~~~~~~~Request~~~~~~~~~~\n" << request_buffer << std::endl;
-
-    // Create response
-    // Echo connection request, pasting in the request after headers,
-    // then the double CRLF at the bottom
-    const std::string res_header = "HTTP/1.1 200 OK\nContent-Type: text/plain\n\n";
-    std::size_t header_size = res_header.size();
-    std::size_t res_size = header_size + bytes_read;
-
-    char response_buffer[res_size+1];
-    res_header.copy(response_buffer, header_size);
-    std::memcpy(&response_buffer[header_size], request_buffer, bytes_read);
-    response_buffer[res_size] = '\0';
-
-    std::cout << "~~~~~~~~~~Response~~~~~~~~~~\n" << response_buffer << std::endl;
-    boost::asio::write(socket,
-                      boost::asio::buffer(response_buffer, res_size));
-
+    boost::shared_ptr<boost::asio::ip::tcp::socket>
+        socket(new boost::asio::ip::tcp::socket(io_service_));
+    acceptor_.accept(*socket);
+
+    // Read in request and handle echo response in echoRequestHandler,
+    // returning the number of bytes written back out to the socket.
+    size_t bytes_written = lightning_server::request_handlers::echoRequestHandler(socket);
```

Comment:
Should `echoRequestHandler` return the response buffer rather than the number of bytes written? Since every request handler will eventually want to write back a response, we could handle the response-writing all in one place, after the request-handlers do their processing. 

Then, the flow would be something like: 

* We accept new connection requests
* We give the socket to a dispatch method
* The dispatch method identifies the right handler to call
* The handler reads in data from the socket and constructs a response buffer
* The response buffer is handed back up the call chain
* The main server loop (or a helper method) writes back the response

More importantly, returning the response buffer lets us test against the expected contents of that buffer.